idb/postgres/internal/writer/writer.go

```diff
@@ -172,12 +177,36 @@ func transactionAssetID(txn transactions.SignedTxnWithAD, intra uint64, block *b
 	return assetid
 }
 
-func countTxns(stxnad transactions.SignedTxnWithAD) uint64 {
-	num := uint64(1)
+func addInnerTransactions(stxnad transactions.SignedTxnWithAD, block *bookkeeping.Block, intra uint64, rootTxid string, batch *pgx.Batch) (uint64, error) {
```

Comment:
This function largely seems to be a duplicate of `addTransactions()`. Have you considered unifying these with something like this? `addTransaction(stxnad, inner bool, ...)`

Reply:
Also, I think we should pass a pointer for `stxnad`. This is because the programmer might think that `stxnad` is owned and we can do anything we want with it, when in fact the maps inside are borrowed. We are in a weird state where some data is copied and owned, and some is borrowed. So I think if we pass an object by value, all maps and slices should be copied; otherwise, we should pass a pointer. I realize some code here already passes transactions by value, but going forward I think we need to improve it.

Reply:
I don't think this is bad, so decided to stick with something that worked. I tried to unify them, but it is tricky. Specifically:
1) the logic has slight differences.
2) we still need to iterate `Payset` and `InnerTxns`, so it is tricky.